ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+                        continue;
+                    }
+                    // ETFs have a special exchange tag associated with them.
+                    // Sometimes, we have an empty exchange for some tickers. Let's try and map them first before giving up on them.
+                    // Example: SPOT (Spotify) on 2018-10-01 has a missing exchange even though it has already IPO'd
+                    if (ticker.Exchange != "NASDAQ" && ticker.Exchange != "NYSE" && ticker.Exchange != "ETF" && !string.IsNullOrEmpty(ticker.Exchange))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} is not in NYSE, NASDAQ, or is not an ETF. Skipping");
+                        continue;
+                    }
+                    // We can't write these tickers in Windows, so we'll skip them
+                    if (_isWindows && _forbiddenTickers.Contains(ticker.Symbol.Value))
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Ticker {ticker.Symbol.Value} is invalid in Windows. Skipping");
+                        continue;
+                    }
+
+                    var tickerPath = new DirectoryInfo(Path.Combine(_destinationDirectory.FullName, ticker.Symbol.Value.ToLowerInvariant()));
+                    tickerPath.Create();
+
+                    var referenceFilePath = Path.Combine(tickerPath.FullName, $"{date:yyyMMdd}.csv");
+                    var tickerReferences = new HashSet<string>();
+
+                    if (File.Exists(referenceFilePath))
+                    {
+                        tickerReferences = File.ReadAllLines(referenceFilePath)
+                            .ToHashSet();
+                    }
+
+                    tickerReferences.Add($"{article.Id}.json");
+
+                    var referenceFileTemp = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.csv");
+                    var finalFileExists = File.Exists(referenceFilePath);
+                    var existingReferenceFileBackupPath = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.csv");
+
+                    Log.Trace($"BenzingaDataConverter.WriteToFile(): Creating reference temp file: {referenceFileTemp}");
+                    File.WriteAllLines(referenceFileTemp, tickerReferences.OrderBy(x => x));
+
+                    if (finalFileExists)
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Moving existing reference file to backup location: {existingReferenceFileBackupPath}");
+                        File.Move(referenceFilePath, existingReferenceFileBackupPath);
+                    }
+
+                    try
+                    {
+                        Log.Trace($"BenzingaDataConverter.WriteToFile(): Writing reference to {referenceFilePath}");
+                        File.Move(referenceFileTemp, referenceFilePath);
+                    }
+                    catch (Exception e)
```

Comment:
_super_ nit -- typically in this codebase we use `error` when handling exceptions.